packages/components/src/FilterDatePicker/subcomponents/FilterDatePickerField/FilterDatePickerField.tsx

```diff
@@ -78,60 +79,103 @@ export const FilterDatePickerField = ({
+  const [state, dispatch] = useReducer(filterDatePickerFieldReducer, {
+    selectedDate,
+    inputValue:
+      transformDateToInputValue(selectedDate, disabledDays, locale) || "",
+    startMonth:
+      selectedDate && !isInvalidDate(selectedDate)
+        ? selectedDate
+        : defaultMonth || new Date(),
+  })
 
   const handleDateChange = (date: Date | undefined): void => {
     onDateChange(date)
   }
 
-  const dateValidation = useSingleDateValidation({
-    disabledDays,
-    validationMessage,
-    onValidate,
-  })
-
-  const validateDate = (date: Date | undefined): Date | undefined =>
-    dateValidation.validateDate({ date, inputValue: inputDateValue })
-
   const inputDateHandlers = useDateInputHandlers({
     locale,
     disabledDays,
-    setInputValue: setInputDateValue,
+    setInputValue: value => {
+      dispatch({
+        type: "update_input_field",
+        inputValue: value as string,
+      })
+    },
     onDateChange: date => {
-      const newDate = validateDate(date)
-      handleDateChange(newDate)
-
-      if (newDate) {
-        setStartMonth(newDate)
+      // Because the input value is being tracked in the state as the user types
+      // we can reliably utilise state.inputValue as part of the validation message,
+      // which is triggered by `onBlur` function
+      const newDate = validateDate(date, state.inputValue)
+
+      dispatch({
+        type: "update_selected_date",
+        date: newDate,
+      })
+
+      // Only provide consumers with a valid date to the `onDateSubmit` function
+      if (newDate && !isInvalidDate(newDate)) {
         onDateSubmit?.(newDate)
       }
+
+      handleDateChange(newDate)
     },
     ...inputProps,
   })
 
   const handleCalendarSelect: CalendarSingleProps["onSelect"] = date => {
-    const newDate = validateDate(date)
-    setInputDateValue(transformDateToInputValue(newDate))
+    // Transforming the date to an InputValue and validating the date with the result
+    // can operate in this order because we are guaranteed a valid date from the calendar.
+    //
+    // In all other scenarios, this won't work as the return string from an invalid date
+    // would be "Invalid Date" and not the actual value entered by the user since we want
+    // the final error message to use the typed word from the user.
+    const inputValue = transformDateToInputValue(date, disabledDays, locale)
+    const newDate = validateDate(date, inputValue)
+
+    dispatch({
+      type: "update_selected_date",
+      date: newDate,
+    })
+
+    dispatch({
+      type: "update_input_field",
+      inputValue,
+    })
+
     handleDateChange(newDate)
-    onDateSubmit?.(newDate)
+
+    // We only want to send valid dates back to the consumer
+    // but the calendar can also clear the vale
+    if (newDate && !isInvalidDate(newDate)) {
+      onDateSubmit?.(newDate)
+    } else {
+      onDateSubmit?.(undefined)
```

Comment:
Yeah, the goal is to only send a valid date which includes the custom validation sent through by consumers. Technically without this condition, if someone were to setup a validation rule saying "no dates in the past" then someone selected a date in the past, it would still trigger `onDateSubmit`.

After explaining this, I don't think there's a test to confirm this behaviour lol

Reply:
Decided this is out of scope and to keep existing behaviour